### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by pipe2442**
On this project, we used Jest to create unit-tests for our tic-tac-toe project. We also made use of Webpack and configure it in order to run our CSS styling and JS files.

Finally, we created testing files and generated multiple unit tests (10+) with the purpose to verify if everything works as planned.

Project Milestones
- Create multiple unit tests for the recently created modules.
- Adding and configuring Webpack

General requirements:
 Add a descriptive README file to your repositories as described.
 Use meaningful names for your Pull Requests and add a summary of its content as described.
 Use descriptive commit messages as described.
 Do not commit files that are not related to the project. Instead, add them to .gitignore file.
 Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

Git Flow:
 Use a specific branch for each feature.
 Use a descriptive name for your branches.
 Use Git to validate your code.

### CODE REVIEW

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️

Hello @pipe2442,

Good job so far with your app testing, however, you are missing one small detail:

- [x] Create separate test file for each class or module. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/test_each_module_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/70449618/)) (Currently you have just index.test.js with all the functions in there, please divide the tests like: startGame.test.js for example and so on)

Please correct those issues and send another review request, good luck, and well done!

**Comment by pipe2442** [comment]
Thanks for the review @e71az the changes were implemented!

File: test/player.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const { Player } = require('../src/index');
```
Comment (rindrajosia) [inline test/player.test.js:1]: [OPTIONNAL] Use import instead of require.

File: test/reload.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const { reload } = require('../src/index');
```
Comment (rindrajosia) [inline test/reload.test.js:1]: [OPTIONNAL] Use import instead of require.

File: test/startGame.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const { startGame } = require('../src/index');
```
Comment (rindrajosia) [inline test/startGame.test.js:1]: [OPTIONNAL] Use import instead of require.

File: test/validateForm.test.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const { validateForm } = require('../src/index');
```
Comment (rindrajosia) [inline test/validateForm.test.js:1]: [OPTIONNAL] Use import instead of require.

File: test/hiddeTable.test.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const { hiddeTable } = require('../src/index');
```
Comment (rindrajosia) [inline test/hiddeTable.test.js:1]: [OPTIONNAL] Use import instead of require.

File: src/index.js [hunk 6]
```diff
@@ -96,4 +100,8 @@ startGame = () => {
     cells[i].addEventListener('click', marker);
   }
   document.getElementById('win').innerHTML = `<h1>It's ${player2.name.value} turn</h1>`;
+};
+
+module.exports = {
+  Player, startGame, reload, validateForm, hiddeTable,
 };
```
Comment (rindrajosia) [inline src/index.js:107]: [OPTIONNAL] Use export instead of module.exports.

File: src/index.js [hunk 7]
```diff
@@ -61,7 +65,7 @@ startGame = () => {
       } else {
         document.getElementById('win').innerHTML = `<h1>${player1.name.value} wins</h1>`;
       }
-      document.getElementById('game_table').style.visibility = 'hidden';
+      hiddeTable();
       document.getElementById('win').style.margin = '-30rem 0 0 0';
```
Comment (rindrajosia) [inline src/index.js:69]: |OPTIONNAL] Keep all the styling logic in CSS files

File: src/index.js [hunk 8]
```diff
@@ -86,7 +90,7 @@ startGame = () => {
     turnNumber += 1;
     if (turnNumber > 8) {
       document.getElementById('win').innerHTML = '<h1>It\'s a tie!!!</h1>';
-      document.getElementById('game_table').style.visibility = 'hidden';
```
Comment (rindrajosia) [inline src/index.js:89]: |OPTIONNAL] Keep all the styling logic in CSS files

File: src/index.js [hunk 9]
```diff
@@ -86,7 +90,7 @@ startGame = () => {
...
+      hiddeTable();
       document.getElementById('win').style.margin = '-30rem 0 0 0';
```
Comment (rindrajosia) [inline src/index.js:94]: |OPTIONNAL] Keep all the styling logic in CSS files

**Review by rindrajosia — APPROVED** [review]
STATUS: APPROVED :tada:
 Weldone for implementing the suggested changes. Your project is approved, kindly merge and prepare for your next task

To Highlight: :angel:

- Separate test file for each class or module was added. :+1: 

Nice to have [OPTIONAL] :

- [ ] As you are using ES6 syntax, use `import` and `export` instead of `require` or `module.exports`.
- [ ] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright).
- [ ] It will perfect if you add more test units. The aim is to make sure if your APP works as expected.

**Pull request merged**